
In kernel 3.12 drm_gem_private_object_init is void

Signed-off-by: Thomas Backlund <tmb@mageia.org>

--- NVIDIA-Linux-x86_64-319.60/kernel/nv-drm.c.orig	2013-09-26 00:02:38.000000000 +0300
+++ NVIDIA-Linux-x86_64-319.60/kernel/nv-drm.c	2013-10-13 18:38:55.689152017 +0300
@@ -201,12 +201,16 @@ RM_STATUS NV_API_CALL nv_alloc_os_descri
     memset(&nv_obj->base, 0, sizeof(nv_obj->base));
     nv_obj->pages = pages;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 12, 0)
+    drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
+#else
     ret = drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
     if (ret)
     {
         status = RM_ERR_OPERATING_SYSTEM;
         goto done;
     }
+#endif
 
     ret = drm_gem_handle_create(file_priv, &nv_obj->base, handle);
     if (ret)
