
Fix build with 3.11 series kernels.

Signed-off-by: Thomas Backlund <tmb@mageia.org>

diff -Nurp NVIDIA-Linux-x86_64-319.32.orig/kernel/nv-linux.h NVIDIA-Linux-x86_64-319.32/kernel/nv-linux.h
--- NVIDIA-Linux-x86_64-319.32.orig/kernel/nv-linux.h	2013-08-01 15:37:03.324309928 +0300
+++ NVIDIA-Linux-x86_64-319.32/kernel/nv-linux.h	2013-08-01 15:41:46.990716198 +0300
@@ -1049,7 +1049,11 @@ static inline int nv_execute_on_all_cpus
 #endif
 
 #if !defined(NV_VMWARE)
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 11, 0))
 #define NV_NUM_PHYSPAGES                num_physpages
+#else
+#define NV_NUM_PHYSPAGES                totalram_pages
+#endif
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
