--- NVIDIA-Linux-x86_64-295.20/kernel/conftest.sh.orig	2012-02-07 07:07:35.000000000 +0200
+++ NVIDIA-Linux-x86_64-295.20/kernel/conftest.sh	2012-03-06 16:24:45.565709044 +0200
@@ -126,7 +126,7 @@ build_cflags() {
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS -I$HEADERS $AUTOCONF_CFLAGS"
 
     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include -I$SOURCES/arch/x86/include/generated"
     elif [ "$ARCH" = "ARMv7" ]; then
         CFLAGS="$CFLAGS -I$SOURCES/arch/arm/include"
     fi
